crates/broker/src/order_picker.rs

```diff
@@ -867,10 +892,43 @@ where
 
                         let picker_clone = picker.clone();
                         let task_cancel_token = cancel_token.child_token();
+                        let order_id = order.id();
+
+                        // Track this task as active
+                        {
+                            let mut active_tasks = picker_clone.active_tasks.lock().await;
+                            active_tasks.insert(order_id.clone(), order.clone());
```

Comment:
wait, why do you need the order itself? Doesn't seem like it's used for anything other than logging the id, which is the key?

Reply:
I did this just so we can log using the Display impl of the order, which includes the useful logging around when the orders expire.